misc/bazel/pkg.bzl

```diff
@@ -248,140 +163,299 @@ def _get_zip_filename(name_prefix, kind):
     if kind == "arch":
         return name_prefix + "-" + _detect_platform() + ".zip"  # using + because there's a select
     else:
-        return "%s-generic.zip" % name_prefix
+        return "%s-common.zip" % name_prefix
+
+def _codeql_pack_info_impl(ctx):
+    zips_to_prefixes = {}
+    for zip_target, prefix in ctx.attr.extra_zips.items():
+        for zip in zip_target.files.to_list():
+            zips_to_prefixes[zip] = prefix
+    return [
+        DefaultInfo(files = depset(zips_to_prefixes.keys(), transitive = [ctx.attr.src[DefaultInfo].files])),
+        _CodeQLPackInfo(arch_overrides = ctx.attr.arch_overrides, files = ctx.attr.src[PackageFilegroupInfo], zips = _ZipInfo(zips_to_prefixes = zips_to_prefixes), pack_prefix = ctx.attr.prefix),
```

Comment:
cosmetic nit: maybe split this into multiple lines, seems like a pretty long one